src/GiraffeGenerator.Core/CodeGenNullChecking.fs

```diff
@@ -0,0 +1,212 @@
+            let value = "value"
+            let mapCheckers = "mapCheckers"
+            letDecl false checkForUnexpectedNullsName [checkers; errType; value] None
+            ^ letOrUseComplexParametersDecl mapCheckers (Pats [SynPat.Paren(tuplePat ["typeName"; "path"; "accessor"], r)])
+                (
+                     letOrUseComplexParametersDecl "v" (Pats [SynPat.Wild(r)])
+                        (
+                            app (identExpr CodeGenErrorsDU.errInnerModelBindingUnexpectedNull)
+                                (
+                                    recordExpr
+                                        [
+                                            "TypeName", identExpr "typeName"
+                                            "PropertyPath", identExpr "path"
+                                        ]
+                                )
+                        )
+                        (
+                            let tryEnumerate =
+                                app (identExpr "accessor") (identExpr value)
+                                ^|> Seq.filterExpr _id
+                                ^|> Seq.mapExpr (identExpr "v")
+                            let clausePat = SynPat.IsInst(synType "System.NullReferenceException", r)
+                            let clauseBody = app (identExpr "v") (tupleExpr []) ^|> app (longIdentExpr "Seq.replicate") (constExpr (SynConst.Int32 1))
+                            SynExpr.TryWith(tryEnumerate,r,[Clause(clausePat,None,clauseBody,r,DebugPointForTarget.Yes)],r,r,DebugPointAtTry.Yes r, DebugPointAtWith.Yes r)
+                        )
+                )
+                (
+                    identExpr checkers
+                    ^|> Seq.collectExpr (identExpr mapCheckers)
+                    ^|> longIdentExpr "Seq.toArray"
+                    ^|> lambda (simplePats[simplePat "v"])
+                        (
+                            ifElseExpr (app (appI (identExpr "op_Equals") (longIdentExpr "v.Length")) (constExpr (SynConst.Int32 0)))
+                                (app (identExpr "Ok") (identExpr value))
+                                (
+                                    ifElseExpr
+                                        (app (appI (identExpr "op_GreaterThan") (longIdentExpr "v.Length")) (constExpr (SynConst.Int32 1)))
+                                        (identExpr "v" ^|> identExpr CodeGenErrorsDU.errInnerCombined)
+                                        (identExpr "v" ^|> longIdentExpr "Array.head")
+                                    ^|> identExpr errType
+                                    ^|> identExpr "Error"
+                                )
+                        )
+                )
+        checkForUnexpectedNulls
+    ]
+
+type private Modifier =
+    | Arr
+    | Opt
+module private Modifiers =
+    let private notNull = Seq.mapExpr (longIdentExpr nullReferenceToOption) ^|> Seq.chooseExpr _id
+    let rec mapExprToSeqOfB currentMods expr =
```

Comment:
few comments what is this would be nice to have!

Reply:
dropped null-checking as unstable and off-topic